File: Server/ShatterWorldsAPI/shatter-worlds-api/src/main/java/shatter/worlds/api/character/Character.java
package shatter.worlds.api.character;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import shatter.worlds.api.character.attributes.Attributes;
import shatter.worlds.api.character.classes.basic.BasicClass;
import shatter.worlds.api.character.classes.prestige.PrestigeClass;
import shatter.worlds.api.player.Player;

import javax.persistence.*;

@AllArgsConstructor
@NoArgsConstructor
@Entity(name = "CHARACTER")
@Getter
@Setter
public class Character {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "SequenceCharacterId")
    @SequenceGenerator(name = "SequenceCharacterId", sequenceName = "CHARACTER_SEQ", allocationSize = 1)
    private Long characterId;

    @ManyToOne
    @JoinColumn( name = "player_owner_id" )
    private Player playerOwner;

    private String name;

    private String race;

    @ManyToOne
    @JoinColumn( name = "basic_class_id" )
    private BasicClass basicClass;

    private int level;

    private int experiencePoints;


    public Character(Player playerOwner, String name, String race, BasicClass basicClass,  int level, int experiencePoints) {
        this.playerOwner = playerOwner;
        this.name = name;
        this.race = race;
        this.basicClass = basicClass;
        this.level = level;
        this.experiencePoints = experiencePoints;
    }
}

File: ShatterWorlds/Assets/System/Out-Battle/Scripts/Model/Character.cs
using System;

namespace outBattle
{
    [Serializable]
    public class Character
    {
        public int characterId;

        public Player player;

        public string name;

        public string race;

        public BasicClass basicClass;

        public int level;

        public int experiencePoints;

        public Attributes attributes;

        public Character(string name, string race, BasicClass basicClass, int level, int experiencePoints,
            Attributes attributes)
        {
            this.name = name;
            this.race = race;
            this.basicClass = basicClass;
            this.level = level;
            this.experiencePoints = experiencePoints;
            this.attributes = attributes;
        }

        public Character(int characterId, Player player, string name, string race, BasicClass basicClass, int level,
            int experiencePoints, Attributes attributes)
        {
            this.characterId = characterId;
            this.player = player;
            this.name = name;
            this.race = race;
            this.basicClass = basicClass;
            this.level = level;
            this.experiencePoints = experiencePoints;
            this.attributes = attributes;
        }


        public Character(Player player, string name, string race, BasicClass basicClass, int level, int experiencePoints,
            Attributes attributes)
        {
            this.player = player;
            this.name = name;
            this.race = race;
            this.basicClass = basicClass;
            this.level = level;
            this.experiencePoints = experiencePoints;
            this.attributes = attributes;
        }
    }
}
